MiniScopeControlDlg.cpp

```diff
@@ -725,25 +726,40 @@ UINT CMiniScopeControlDlg::msCapture(LPVOID pParam )
 		status = self->msCam.grab();
 		if (status == false) {
 			self->record = false;
-			self->AddListText(L"msCam frame grab error! Recording ended.");
-			break;
+			self->AddListText(L"msCam frame grab error!");
+			continue;
 		}
 		previousTime = currentTime;
 		QueryPerformanceCounter(&currentTime);
 		self->mMSCurrentFPS = 1/(((double)currentTime.QuadPart - previousTime.QuadPart)/self->Frequency.QuadPart);
 		
 		self->msCapFrameTime[self->msWritePos%BUFFERLENGTH] = 1000*((double)currentTime.QuadPart - self->startOfRecord.QuadPart)/self->Frequency.QuadPart;
 		
+
 		status = self->msCam.retrieve(self->msFrame[self->msWritePos%BUFFERLENGTH]);
 		if (status == false) {
-			self->record = false; //Commented out to not end recording Daniel 11_10_2015
+			//self->record = false;
 			self->mMSDroppedFrames++; //Added frame drop tracker Daniel 11_10_2015
-			self->AddListText(L"msCam frame retrieve error! Recording ended.");
+			self->AddListText(L"msCam frame retrieve error!");
 			//self->dFrameDrop.ShowWindow(SW_SHOW);//popup dialog box Daniel 11_10_2015
 			cv::imshow("msCam",droppedFrameImage);
-			break; //removed break Daniel 11_10_2015
+			if (self->mMSDroppedFrames > 0) {
+				self->AddListText(L"reconnecting");
+				self->msCam.release();
+				self->msCam.open(self->mScopeCamID);
+			}
+			continue;
 		}
 		else {//Added else Daniel 11_10_2015
+			if (self->mMSDroppedFrames > 0 || self->mMSCurrentFPS < self->mMSFPS / 2.0) {
```

Comment:
Hey @phildong, I don't think self->mMSDroppedFrames ever gets reset and instead just increments every time self->msCam.retreive() fails. From my understanding, self->mMSDroppedFrames > 0 is check to see if the Miniscope is recovering from a recent disconnect. If this is the case then wouldn't you want to be checking the self->mMSDroppedFrames incremented up instead of it just being > 0?

Reply:
Hi @daharoni , I put mMSDroppedFrames reset on recovery (next valid frame), see line 760 below. In this case I think `self->mMSDroppedFrames > 0` is almost equivalent to `status == false`, the reason I didn't just use `status` is this way we can easily make the program tolerate multiple dropped frames later, though that might not be a good idea since each dropped frame implies lose of at least 1 sec of data.

Reply:
Thanks for the info!